src/main/scala/org/broadinstitute/dsde/firecloud/service/NihService.scala

```diff
@@ -72,78 +74,88 @@ trait NihService extends LazyLogging {
   val thurloeDao: ThurloeDAO
   val googleDao: GoogleServicesDAO
 
+  val nihWhitelists: Set[NihWhitelist] = FireCloudConfig.Nih.whitelists
+
   def getNihStatus(userInfo: UserInfo): Future[PerRequestMessage] = {
     thurloeDao.getProfile(userInfo) flatMap {
       case Some(profile) =>
         profile.linkedNihUsername match {
           case Some(_) =>
-            rawlsDao.isDbGapAuthorized(userInfo) map { isDbGapAuthorized =>
-              RequestComplete(NihStatus(profile.copy(linkExpireTime = Option(profile.linkExpireTime.getOrElse(0L))), Some(isDbGapAuthorized)))
+            Future.traverse(nihWhitelists) { whitelistDef =>
+              rawlsDao.isGroupMember(userInfo, whitelistDef.groupToSync).map(isMember => NihWhitelistStatus(whitelistDef.name, isMember))
+            }.map { whitelistMembership =>
+              RequestComplete(NihStatus(profile.copy(linkExpireTime = Option(profile.linkExpireTime.getOrElse(0L))), whitelistMembership))
```

Comment:
OK but `None` would express that just as well, right?  It seems to me that this logic should live in the consumer of this message instead of here.  If it really does need to work this way, then `NihStatus.linkExpireTime` should not be an `Option`.